### ISSUE TITLE
Event not found in Podman 3.4.4 on Ubuntu Jammy

### ISSUE THREAD

**Issue by KEO-RSP** [containers/podman#16217]
/kind bug

**Description**

We are running Podman 3.4.4 on Ubuntu Jammy on a ThreadRipper server with 128 CPUs and 64 GB RAM. We are facing an issue we didn't face on the same hardware with Podman 3.4.2 on Focal. We are speaking about a 64-bit CPU here.

Podman 3.4.4 was installed directly via apt because it's part of Jammy.
Podman 3.4.2 was installed as described [here](https://www.how2shout.com/linux/how-to-install-podman-on-ubuntu-20-04-lts-focal-fossa/) because it is not regularly part of Focal.

We run a lot of containers in parallel and quickly after each other because this is a CI server where we cross-compile lots of things.

The issue we are facing is that we sometimes see the following in Jenkins job console output:

```
time="2022-10-17T11:04:01+02:00" level=error msg="Could not retrieve exit code from event: died not found: unable to find event"
  
[ERROR] *** Configure run failed with error 32512
  
Build step 'Execute shell' marked build as failure`
```

The first line comes directly from podman. The second line shows how the Jenkins job ends. The third line is pure Jenkins output.

Here we have three questions:

1. Why does that message appear on standard output and doesn't go to some Podman log facility? It also appears in the terminal during reproduction of the issue
2. What is exit code 32512? Note that this is 0x7F00 whereas GetContainerExitCode returns 127 in case of error which is 0x7F. (Might get mixed up by Jenkins - potentially not part of the issue)
3. How can it happen that the exit code of the container cannot be determined? Has it ever occured that events get dropped? Is this fixed in newer versions? I couldn't find a suitable issue here on Github or via Google so far.

**Steps to reproduce the issue:**

1. We started the same container with the same command over and over. It failed in roughly one out of five cases. However, upon another attempt at reproducing the issue when there were different loads on the overall system, it failed only in less cases, but we still managed to make it fail.

**Describe the results you received:**

Even though the stuff that ran in the container finished successfully, the container exited with an exit code other than zero, namely 127 in this case. (Let's ignore the number mangling towards 32512) A podman-specific log message appeared on the console in addition.

**Describe the results you expected:**

Because the stuff that ran in the container finished successfully, the container should exit with an exit code of zero. The log message should go to the central podman log facility.

**Additional information you deem important (e.g. issue happens only occasionally):**

The behaviour happens sporadically. We have not yet determined an actual reason. This looks like a racing to us.

**Output of `podman version`:**

```
Version:      3.4.4
API Version:  3.4.4
Go Version:   go1.17.3
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.25, commit: unknown'
  cpus: 128
  distribution:
    codename: jammy
    distribution: ubuntu
    version: "22.04"
  eventLogger: journald
  hostname: rocinante
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1002
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1002
      size: 1
    - container_id: 1
      host_id: 231072
      size: 65536
  kernel: 5.15.0-50-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 49214599168
  memTotal: 67247951872
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'
    path: /usr/bin/crun
    version: |-
      crun version 0.17
      commit: 0e9229ae34caaebcb86f1fde18de3acaf18c6d9a
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1002/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.0.1
      commit: 6a7b16babc95b6a3056b33fb45b74a6f62262dd4
      libslirp: 4.6.1
  swapFree: 137011392512
  swapTotal: 137438949376
  uptime: 122h 54m 47.24s (Approximately 5.08 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries: {}
store:
  configFile: /home/devops/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: 'fuse-overlayfs: /usr/bin/fuse-overlayfs'
      Version: |-
        fusermount3 version: 3.10.5
        fuse-overlayfs: version 1.7.1
        FUSE library version 3.10.5
        using FUSE kernel interface version 7.31
  graphRoot: /home/devops/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/1002/containers
  volumePath: /home/devops/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.17.3
  OsArch: linux/amd64
  Version: 3.4.4
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
devops@rocinante:~$ apt list podman
Listing... Done
podman/jammy,now 3.4.4+ds1-1ubuntu1 amd64 [installed]
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Test with latest version of Podman: no
Read the troubleshooting guide: yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Nothing specific. Please feel free to ask for details :)

**Comment by baude** [comment]
Are you willing to update to Podman 4 and try it there?

**Comment by KEO-RSP** [comment]
We would certainly be willing to try that. There seems to be no obvious way to get Podman 4+ onto a Jammy installation, which approach would you recommend? Which set of software would we need to update other than just the podman binary?

**Comment by KEO-RSP** [comment]
> There seems to be no obvious way to get Podman 4+ onto a Jammy installation, which approach would you recommend?

Never mind, we will go with the Kubic repositories.

**Comment by baude** [comment]
Are your results better?

**Comment by KEO-cls-eedi** [comment]
> Are your results better?

We do. We now have:

```
Client:       Podman Engine
Version:      4.3.0
API Version:  4.3.0
Go Version:   go1.18.1
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

installed via https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_22.04.

We have repeatedly triggered a lot of parallel builds and so far the issue has not come up again. We are going to stick to this setup for now and provide updates if anything goes wrong.

It would be ideal to find Podman 4 in the official Ubuntu repositories at some point. Do you have any idea whether this is planned or should I directly ask the maintainers as listed at https://packages.ubuntu.com/jammy/podman?

**Comment by KEO-RSP** [comment]
We do. We now have:

```
Client:       Podman Engine
Version:      4.3.0
API Version:  4.3.0
Go Version:   go1.18.1
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

installed via https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_22.04.

We have repeatedly triggered a lot of parallel builds and so far the issue has not come up again. We are going to stick to this setup for now and provide updates if anything goes wrong.

It would be ideal to find Podman 4 in the official Ubuntu repositories at some point. Do you have any idea whether this is planned or should I directly ask the maintainers as listed at https://packages.ubuntu.com/jammy/podman?

**Comment by vrothberg** [comment]
Thanks for checking!

> It would be ideal to find Podman 4 in the official Ubuntu repositories at some point. Do you have any idea whether this is planned or should I directly ask the maintainers listed at https://packages.ubuntu.com/jammy/podman?

@siretart PTAL

**Comment by rhatdan** [comment]
This is not a Upstream issue, conversation can continue here.
